app/controllers/bookmarks_controller.rb

```diff
@@ -32,6 +32,21 @@ def edit
 	  authorize @bookmark
 	end
 
+  def update
+    @bookmark = Bookmark.find(params[:id])
+    @topic = Topic.find(params[:topic_id])
```

Comment:
Consider applying suggested code changes in `edit` to these first 2 lines of `update`.